Sources/PackagePlugin/CMakeLists.txt

```diff
@@ -7,60 +7,63 @@
-    target_link_options(PackagePlugin PRIVATE
-        "SHELL:-Xlinker -install_name -Xlinker @rpath/libPackagePlugin.dylib")
+  set(SWIFT_INTERFACE_PATH ${CMAKE_BINARY_DIR}/pm/PluginAPI/PackagePlugin.swiftinterface)
+  target_compile_options(PackagePlugin PUBLIC
+    $<$<COMPILE_LANGUAGE:Swift>:-emit-module-interface-path$<SEMICOLON>${SWIFT_INTERFACE_PATH}>)
+  target_link_options(PackagePlugin PRIVATE
+    "SHELL:-Xlinker -install_name -Xlinker @rpath/libPackagePlugin.dylib")
 endif()
 
 set_target_properties(PackagePlugin PROPERTIES
-    Swift_MODULE_NAME PackagePlugin
-    Swift_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/pm
-    INSTALL_NAME_DIR \\@rpath
-    OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/pm
-    OUTPUT_NAME PackagePlugin
-    ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/pm
-    LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/pm
-    RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/pm
+  Swift_MODULE_NAME PackagePlugin
+  Swift_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/pm/PluginAPI
+  INSTALL_NAME_DIR \\@rpath
+  OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/pm/PluginAPI
+  OUTPUT_NAME PackagePlugin
+  ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/pm/PluginAPI
+  LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/pm/PluginAPI
+  RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/pm/PluginAPI
 )
 
 if(NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
-    if(Foundation_FOUND)
-        target_link_libraries(PackagePlugin PRIVATE
-            Foundation)
-    endif()
-    target_link_options(PackagePlugin PRIVATE
-        "SHELL:-no-toolchain-stdlib-rpath")
-    set_target_properties(PackagePlugin PROPERTIES
-        BUILD_WITH_INSTALL_RPATH TRUE
-        INSTALL_RPATH "$ORIGIN/../../$<LOWER_CASE:${CMAKE_SYSTEM_NAME}>")
+  if(Foundation_FOUND)
+    target_link_libraries(PackagePlugin PRIVATE
+      Foundation)
+  endif()
+  target_link_options(PackagePlugin PRIVATE
+    "SHELL:-no-toolchain-stdlib-rpath")
+  set_target_properties(PackagePlugin PROPERTIES
+    BUILD_WITH_INSTALL_RPATH TRUE
+    INSTALL_RPATH "$ORIGIN/../../$<LOWER_CASE:${CMAKE_SYSTEM_NAME}>")
 endif()
 
 if(CMAKE_HOST_SYSTEM_NAME STREQUAL Darwin)
-    install(FILES
-        ${CMAKE_BINARY_DIR}/pm/PackagePlugin.swiftinterface
-        ${CMAKE_BINARY_DIR}/pm/PackagePlugin.swiftdoc
-        DESTINATION lib/swift/pm
-    )
+  install(FILES
+    ${CMAKE_BINARY_DIR}/pm/PluginAPI/PackagePlugin.swiftinterface
+    ${CMAKE_BINARY_DIR}/pm/PluginAPI/PackagePlugin.swiftdoc
+    DESTINATION lib/swift/pm/PluginAPI
+  )
 else()
-    install(FILES
-        ${CMAKE_BINARY_DIR}/pm/PackagePlugin.swiftmodule
-        ${CMAKE_BINARY_DIR}/pm/PackagePlugin.swiftdoc
-        DESTINATION lib/swift/pm
-    )
+  install(FILES
+    ${CMAKE_BINARY_DIR}/pm/PluginAPI/PackagePlugin.swiftmodule
+    ${CMAKE_BINARY_DIR}/pm/PluginAPI/PackagePlugin.swiftdoc
+    DESTINATION lib/swift/pm/PluginAPI
+  )
 endif()
 
 install(TARGETS PackagePlugin
-    DESTINATION lib/swift/pm)
+  DESTINATION lib/swift/pm/PluginAPI)
```

Comment:
With PackageDescription no longer living in a directory named after the version number to keep the library apart from the PackagePlugin library, I put each subsystem's API into a separate subdirectory here named after the subsystem for which this is an API.  The plugin one is called `PluginAPI`, and I can imagine that we'd have additional ones as needed.